Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -441,6 +457,9 @@ public HdAddress GetOrCreateChangeAddress(HdAccount account)
                 // get address to send the change to
                 changeAddress = account.GetFirstUnusedChangeAddress();
 
+                // persist the address to the wallet files
+                this.SaveWallets();
```

Comment:
` this.SaveWallets();` is locking maybe we take it out of the current lock?

Also bellow I think the logic I introduced is wrong 
```changeAddress == null```
It never be null because of this line
```  changeAddress = account.InternalAddresses.First(a => a.Address == accountAddress);```
I suggest we remove that `if` so it will just look as following

```
 this.LoadKeysLookupLock();
 this.SaveWallets();
```

Reply:
Fixing this. Will commit shortly.

Reply:
I have a test failing on LoadKeysLookupLock(), I think this test https://github.com/stratisproject/StratisBitcoinFullNode/blob/be520ab2580da771545c630369ee2e35b0d903c0/Stratis.Bitcoin.Features.Wallet.Tests/WalletManagerTest.cs#L758 addresses need to get reworked. Just to confirm with you that it's the test that needs to get fixed and the logic above is correct?

Reply:
I had to refactor the above test to make it pass. The change is part of this PR.